lib/languages/js/source-code/source-code.js

```diff
@@ -58,6 +58,10 @@ const CODE_PATH_EVENTS = [
  * @private
  */
 function validate(ast) {
+    if (!ast) {
+        throw new Error(`AST is ${ast}.`);
```

Comment:
I'm not sure this is clear enough. It's stating what is but doesn't give any indication as to why it's undesirable. Something like this may be better:

```suggestion
        throw new TypeError(`Unexpected empty AST. (${ast})`);
```